### PR TITLE
API - 2988:  Update Accessibility Props for Hover Image

### PR DESCRIPTION
**Pull request by TS000**
### Description
[Jira ticket](https://vajira.max.gov/browse/API-2988)
Added alt tag props to dynamic image. Added new prop portal to News. 

### Process
Test for HoverImage now includes the proper props check with addition of the new alt tag prop.
- [✔️ ] Tests added or updated for change
- [  n/a ] Docs updated
- [ ✔️ ] Accessibility concerns have been considered and addressed

### Requested feedback
There are some greater accessibility things that AXE caught but I wanted to keep within the bounds of the ticket.

### CODE REVIEW

File: src/components/hoverImage/HoverImage.tsx [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
 
+  const setProps = (): void => {
+    setAlt(props.imageAlt);
+    setSrc(props.hoverImagePath);
+  };
+
   return (
     <img
-      alt=""
+      alt={alt}
```
Comment (mjlumetta) [inline src/components/hoverImage/HoverImage.tsx:20]: can you explain why we're making this change? i would've thought that this alt attribute was empty on purpose to mark the image as presentational. the reason i could see it being necessary in context would be that it's nested within a link. however, the alt doesn't actually have any effect unless you also remove the `role="presentation"` attribute. so those are my questions. if we don't want to change this image being presentational, i wonder whether this ticket is necessary. haven't tested yet, will see if that gives me any insight.

i wonder whether we might also be open to getting a quick look from a designer on this one. i'm not sure how much good duplicating the link as an icon helps the user experience? anyway.

File: src/components/hoverImage/HoverImage.tsx [hunk 2]
```diff
@@ -1,20 +1,27 @@
 import * as React from 'react';
 
 interface HoverImageProps {
+  imageAlt: string;
```
Comment (mjlumetta) [inline src/components/hoverImage/HoverImage.tsx:4]: maybe make this optional and default to the empty string if not provided, probably? having an empty alt should be a valid choice, but i also see why we might want to make the consumer pass it explicitly.

File: src/components/hoverImage/HoverImage.tsx [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
   imagePath: string;
   hoverImagePath: string;
 }
 
 const HoverImage = (props: HoverImageProps): JSX.Element => {
+  const [alt, setAlt] = React.useState(props.imageAlt);
   const [src, setSrc] = React.useState(props.imagePath);
 
+  const setProps = (): void => {
+    setAlt(props.imageAlt);
```
Comment (mjlumetta) [inline src/components/hoverImage/HoverImage.tsx:14]: why are we changing the alt here?

**Review by mjlumetta — COMMENTED** [review]
i have a question about the purpose of this PR. i was under the impression that the image was supposed to be presentational on purpose, in which case an empty alt would be the right call. but i could also see having an empty alt inside the link being problematic.

File: src/components/hoverImage/HoverImage.tsx (hunk 1 above)
Comment (TS000) [reply src/components/hoverImage/HoverImage.tsx:20]: The reason for this change was to add an alt tag to the dynamic image. The ticket was made by @alexandervalencia he might have more to say.

File: src/components/hoverImage/HoverImage.tsx (hunk 3 above)
Comment (TS000) [reply src/components/hoverImage/HoverImage.tsx:14]: [A11y](https://www.a11yproject.com/posts/2013-01-14-alt-text/). Essentially since this area changes we wanted a way to dynamically tell screen readers what the image is.

Comment (mjlumetta) [reply src/components/hoverImage/HoverImage.tsx:14]: i guess what i was trying to ask here was about whether the alt is actually dynamic. does the alt change on hover? there are two props for the image source, one for no hover and one for hover. there's one prop for the alt, so i'm wondering whether it would actually change the DOM on hover events or reset the same alt text.

to take a step back, though, does the image conceptually change on hover?

File: src/containers/News.tsx [hunk 4]
```diff
@@ -64,6 +64,7 @@ const MediaItem = ({ item }: { item: NewsItemData }): JSX.Element => {
       <div aria-hidden>
         <a href={item.url} tabIndex={-1}>
           <HoverImage
+            imageAlt={item.title}
```
Comment (gileswells) [inline src/containers/News.tsx:67]: As this image is purely for decoration and not something that includes any specific information about news item it decorates I would argue that the proper alt text here is an empty string because it doesn't add anything of consequence and doubling up on the title would just add frustration for screen reader users.

File: src/components/hoverImage/HoverImage.tsx (hunk 2 above)
Comment (gileswells) [reply src/components/hoverImage/HoverImage.tsx:4]: I concur in allowing this to be optional and having it default to an empty string.

Comment (gileswells) [reply src/components/hoverImage/HoverImage.tsx:4]: Also personally I would prefer to see this property just named `alt` as opposed to the current `imageAlt` so that it uses the more expected prop name.

File: src/components/hoverImage/HoverImage.test.tsx [hunk 5]
```diff
@@ -9,20 +9,21 @@ afterEach(cleanup);
 
 const props = {
   hoverImagePath: 'hover',
+  imageAlt: 'United States',
```
Comment (gileswells) [inline src/components/hoverImage/HoverImage.test.tsx:12]: Making `imageAlt` or `alt` optional would need an additional unit test to cover when the prop is omitted and defaults to an empty string.

Comment (gileswells) [reply src/components/hoverImage/HoverImage.test.tsx:12]: Also currently setting the alt text, while it's being done in this test, isn't actually being confirmed with an `expect()`

**Review by gileswells — CHANGES_REQUESTED** [review]

File: src/containers/News.tsx (hunk 4 above)
Comment (TS000) [reply src/containers/News.tsx:67]: > doubling up on the title would just add frustration for screen reader users.

Is there a better option for the alt text since no alt text is provided?

File: src/components/hoverImage/HoverImage.test.tsx (hunk 5 above)
Comment (TS000) [reply src/components/hoverImage/HoverImage.test.tsx:12]: > Also personally I would prefer to see this property just named alt as opposed to the current imageAlt so that it uses the more expected prop name.

With this direction would it also make more sense to change imagePath to url and hoverImagePath to hoverUrl?

File: src/components/hoverImage/HoverImage.tsx (hunk 2 above)
Comment (TS000) [reply src/components/hoverImage/HoverImage.tsx:4]: > Also personally I would prefer to see this property just named alt as opposed to the current imageAlt so that it uses the more expected prop name.

With this direction would it also make more sense to change imagePath to url and hoverImagePath to hoverUrl?

File: src/components/hoverImage/HoverImage.test.tsx (hunk 5 above)
Comment (TS000) [reply src/components/hoverImage/HoverImage.test.tsx]: made some updates to test

File: src/containers/News.tsx (hunk 4 above)
Comment (TS000) [reply src/containers/News.tsx]: I ended up setting the tag to props.alt and it seems to be picking up a tag 🚀

File: src/components/hoverImage/HoverImage.tsx (hunk 3 above)
Comment (TS000) [reply src/components/hoverImage/HoverImage.tsx]: ended up making the alt tag directly updated my props rather than updating on hover

File: src/components/hoverImage/HoverImage.tsx (hunk 2 above)
Comment (TS000) [reply src/components/hoverImage/HoverImage.tsx]: Naming has been changed and we are now checking for an empty string. I would like some eyes on my test however. I was confused on checking between a string and an empty string since they are both strings.

File: src/components/hoverImage/HoverImage.tsx [hunk 6]
```diff
@@ -10,11 +11,11 @@ const HoverImage = (props: HoverImageProps): JSX.Element => {
 
   return (
     <img
-      alt=""
+      alt={props.alt ? props.alt : ''}
```
Comment (mjlumetta) [inline src/components/hoverImage/HoverImage.tsx:14]: what do you think about `props.alt ?? ''` for the syntax of this?

File: src/containers/News.tsx [hunk 7]
```diff
@@ -28,6 +28,7 @@ interface NewsSection extends DataSection {
 }
 
 export interface NewsItemData {
+  alt: string;
```
Comment (mjlumetta) [inline src/containers/News.tsx:31]: can we remove this, actually? this property doesn't exist in the `news.yml` schema

**Review by mjlumetta — COMMENTED** [review]
so, think this looks good from the code perspective, but my question would be whether or not we want to make this change. because the news media video icons don't actually have alt text, it feels like we're solving for a future situation where the `HoverImage` component is used for something else that does need alt text. so i'm cool with making the change to cover that scenario, but i wanted to make sure we're all on board with making the change for that reason.

File: src/containers/News.tsx (hunk 7 above)
Comment (TS000) [reply src/containers/News.tsx:31]: Yes.

File: src/components/hoverImage/HoverImage.tsx (hunk 6 above)
Comment (TS000) [reply src/components/hoverImage/HoverImage.tsx:14]: I agree it is a bit redundant.

Comment (TS000) [reply src/components/hoverImage/HoverImage.tsx:14]: It actually throws an error if I use your script. I think it's better to leave it as it was.

File: src/containers/News.tsx (hunk 7 above)
Comment (TS000) [reply src/containers/News.tsx:31]: If I remove it from the interface it's no longer available in the component and throws a type error.

Comment (TS000) [reply src/containers/News.tsx:31]: Should I update the yml file associated with this?

**Comment by alexandervalencia** [comment]
Hey @TS000 I was wondering what you thought about writing the component this way to use a default value as opposed to using a ternary to set the alt text. We would also set the prop to optional in the interface:
```js
interface HoverImageProps {
  alt?: string;
  imagePath: string;
  hoverImagePath: string;
}

const HoverImage = ({ alt = '', imagePath, hoverImagePath }: HoverImageProps): JSX.Element => {
  const [src, setSrc] = React.useState(imagePath);

  return (
    <img
      alt={alt}
      role="presentation"
      src={src}
      onMouseEnter={(): void => setSrc(hoverImagePath)}
      onMouseLeave={(): void => setSrc(imagePath)}
    />
  );
};
```

**Comment by alexandervalencia** [comment]
This would solve the issue @mjlumetta was talking about with how the `News` component is currently *not* passing an `alt` prop to the `HoverImage` component

**Review by mjlumetta — COMMENTED** [review]
still have the question about whether there should be functional changes to the site as a result of this PR. for me, the core question is whether the video icons on the News page should change from presentational to functional by way of having a non-empty alt. would like to understand the thinking behind that more clearly - maybe it's worth syncing up on this one?

File: src/components/hoverImage/HoverImage.tsx [hunk 8]
```diff
@@ -1,20 +1,21 @@
...
 
-const HoverImage = (props: HoverImageProps): JSX.Element => {
-  const [src, setSrc] = React.useState(props.imagePath);
+const HoverImage = ({ alt = '', imagePath, hoverImagePath }: HoverImageProps): JSX.Element => {
+  const [src, setSrc] = React.useState(imagePath);
 
   return (
     <img
-      alt=""
-      role="presentation"
```
Comment (mjlumetta) [inline src/components/hoverImage/HoverImage.tsx:14]: i think it is appropriate to conditionally include the presentation role if the alt is empty. i don't think it matters at all in context because it's nested under that `aria-hidden` ancestor, but we might want it elsewhere. i'm okay with this as is given that it's not necessarily applicable at the moment, though.

**Review by mjlumetta — APPROVED** [review]
approved ✅ left a small non-blocking comment about the role question. will re-approve if you choose to make changes, but no objections if you want to merge as is

File: src/components/hoverImage/HoverImage.tsx (hunk 8 above)
Comment (TS000) [reply src/components/hoverImage/HoverImage.tsx:14]: I'm into addressing it the need arises.
